mono/mini/debugger-agent.c

```diff
@@ -4090,6 +4234,9 @@ send_types_for_domain (MonoDomain *domain, void *user_data)
 static void
 send_assemblies_for_domain (MonoDomain *domain, void *user_data)
 {
+#ifdef IL2CPP_MONO_DEBUGGER
+	il2cpp_send_assemblies_for_domain(domain, user_data, emit_assembly_load);
```

Comment:
Are we calling this separately because the code lives in libil2cpp? doesn't seem like much code for it to be forwarded to its own function to me looking at the mono, and I'd rather have the #defined code live next to eachother for comparison purposes personally

Reply:
The code in `il2cpp_send_assemblies_for_domain` needs to have access to types that are defined in C++, so it needs to live in a different translation unit that is compiled with C++ instead of C.